pkg/agent/multicast/mcast_controller.go

```diff
@@ -298,15 +369,30 @@ func (c *Controller) syncGroup(groupKey string) error {
 		return nil
 	}
 	status := obj.(*GroupMemberStatus)
+	memberPorts := make([]uint32, 0)
+	status.mutex.Lock()
+	defer status.mutex.Unlock()
```

Comment:
Is the mutex really needed? I see the object is never updated in-place.

Reply:
I changed to RLock() and RUnLock() since the status is read when getting the OF port and check if it is stale in function `groupIsStale`

Reply:
I mean the whole `status` is never updated so concurrent read/write never happens, does it still need a mutex?

Reply:
Removed.